src/UI/class-recommended-widget.php

```diff
@@ -276,7 +276,7 @@ class="tiny-text" aria-labelledby="<?php echo esc_attr( $this->get_field_id( 'pu
 	public function update( $new_instance, $old_instance ): array {
 		$instance                        = $old_instance;
 		$instance['title']               = trim( wp_kses_post( $new_instance['title'] ) );
-		$instance['published_within']    = (int) trim( $new_instance['published_within'] );
+		$instance['published_within']    = is_int( $new_instance['published_within'] ) ? $new_instance['published_within'] : (int) trim( $new_instance['published_within'] );
```

Comment:
Should there be a test that ensures we capture the logic being added here?

Reply:
Yes, we'll cover it here: https://github.com/Parsely/wp-parsely/issues/399